boot/bootutil/include/bootutil/crypto/ecdsa.h

```diff
@@ -0,0 +1,590 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2023 Arm Limited
+ */
+
+/*
+ * This module provides a thin abstraction over some of the crypto
+ * primitives to make it easier to swap out the used crypto library.
+ *
+ * At this point, the choices are: MCUBOOT_USE_TINYCRYPT, MCUBOOT_USE_CC310,
+ * MCUBOOT_USE_MBED_TLS, MCUBOOT_USE_PSA_CRYPTO. Note that support for
+ * MCUBOOT_USE_PSA_CRYPTO is still experimental and it might not support all
+ * the crypto abstractions that MCUBOOT_USE_MBED_TLS supports. For this
+ * reason, it's allowed to have both of them defined, and for crypto modules
+ * that support both abstractions, the MCUBOOT_USE_PSA_CRYPTO will take
+ * precedence.
+ */
+
+#ifndef __BOOTUTIL_CRYPTO_ECDSA_H_
+#define __BOOTUTIL_CRYPTO_ECDSA_H_
+
+#include <stdint.h>
+#include "mcuboot_config/mcuboot_config.h"
+
+#if defined(MCUBOOT_USE_PSA_CRYPTO) || defined(MCUBOOT_USE_MBED_TLS)
+#define MCUBOOT_USE_PSA_OR_MBED_TLS
+#endif /* MCUBOOT_USE_PSA_CRYPTO || MCUBOOT_USE_MBED_TLS */
+
+#if defined(MCUBOOT_SIGN_EC384) && \
+    !defined(MCUBOOT_USE_PSA_CRYPTO)
+    #error "P384 requires PSA_CRYPTO to be defined"
+#endif
+
+#if (defined(MCUBOOT_USE_TINYCRYPT) + \
+     defined(MCUBOOT_USE_CC310) + \
+     defined(MCUBOOT_USE_PSA_OR_MBED_TLS)) != 1
+    #error "One crypto backend must be defined: either CC310/TINYCRYPT/MBED_TLS/PSA_CRYPTO"
+#endif
+
+#if defined(MCUBOOT_USE_TINYCRYPT)
+    #include <tinycrypt/ecc_dsa.h>
+    #include <tinycrypt/constants.h>
+    #define BOOTUTIL_CRYPTO_ECDSA_P256_HASH_SIZE (4 * 8)
```

Comment:
Could you move the BOOTUTIL_CRYPTO_ECDSA_P256_HASH_SIZE define
to a common macro section just like here?
https://github.com/mcu-tools/mcuboot/pull/1660/files#diff-e3bebec8c109d099a2dc6d5503ae58954a164c647d8831c5adc680fe9c045abdR37

Reply:
Done